src/components/App/App.js

```diff
@@ -1,37 +1,201 @@
+    setSearchIsOpen(false);
+    history.push('/');
+  };
+
+  const handleSearch = (searchWord) => {
+    setSearchWord(searchWord);
+    setLoader(true);
+    setNoArticlesFound(false);
+    setSearchIsOpen(false);
+    newsApi
+      .getArticles(searchWord)
+      .then((articleArray) => {
+        setLoader(false);
+        if (articleArray.articles.length > 0) {
+          setArticleCards(articleArray.articles);
+          articleArray.articles.map((cardObj) => (cardObj.marked = 'false'));
+          setSearchIsOpen(true);
+        } else {
+          setNoArticlesFound(true);
+        }
+      })
+      .catch((err) => {
+        console.log(err);
+        setLoader(false);
+        setSearchIsOpen(false);
+      });
+  };
+
+  const deleteArticle = (article) => {
+    mainApi
+      .deleteArticle(article._id, token)
+      .then(() => {
+        setSavedArticles((articleArray) =>
+          articleArray.filter((a) => a._id !== article._id),
+        );
+        setArticleCards((articleState) =>
+          articleState.map((a) =>
+            a.title === article.title ? { ...a, marked: 'false' } : a,
+          ),
+        );
+      })
+      .catch((err) => console.log(err));
+  };
+
+  const handleSaveCard = (card) => {
+    if (isLoggedIn) {
+      if (card.owner === currentUser._id || card.marked === 'true') {
+        if (card.marked === 'true') {
+          const foundSavedArticle = savedArticles.find(
+            (f) => f.title === card.title,
+          );
+          deleteArticle(foundSavedArticle);
+        } else {
+          deleteArticle(card);
+        }
+      } else {
+        mainApi
+          .saveArticle(token, card, searchWord)
+          .then((newArticle) => {
+            setSavedArticles([...savedArticles, newArticle]);
+            setArticleCards((state) =>
+              state.map((c) =>
+                c.title === card.title ? { ...c, marked: 'true' } : c,
+              ),
+            );
+          })
+          .catch((err) => console.log(err));
+      }
+    } else {
+      handleLoginButton();
+    }
   };
 
   const handleToggleMenu = () => {
     setShowMobileMenu(!showMobileMenu);
   };
 
-  const handleCloseModal = () => {
-    setModalIsOpen(false);
+  const handleLoggedInSavedNewsClick = () => {
+    setIsLoggedInSavedNews(true);
+    handleToggleMenu();
   };
 
-  const handleScreenSize = () => {
-    setScreenWidth(window.innerWidth);
+  const handleAtHomeClick = () => {
+    setIsLoggedInSavedNews(false);
+    handleToggleMenu();
   };
 
+  useEffect(() => {
+    if (token) {
+      mainApi
+        .getUserInfo(token)
+        .then((res) => {
+          setIsLoggedIn(true);
+          setCurrentUser(res.data);
+        })
+        .catch((err) => console.log(err));
+    } else {
+      setIsLoggedIn(false);
+    }
+  }, [token, savedArticles]);
+
+  useEffect(() => {
+    if (token) {
+      mainApi.getUserArticles(token).then((res) => {
+        setSavedArticles(res);
+      });
```

Comment:
You need to `catch` possible errors at the very end of any server request.

Please, correct it everywhere